provider/openstack/provider.go

```diff
@@ -1039,17 +1039,58 @@ func (e *Environ) StartInstance(args environs.StartInstanceParams) (*environs.St
 		args.InstanceConfig.MachineId,
 	)
 
+	waitForActiveServerDetails := func(
+		client *nova.Client,
+		id string,
+	) (server *nova.ServerDetail, err error) {
+		logger.Debugf("waiting for instance %s to start...", id)
+
+		for {
+			server, err = client.GetServer(id)
+			if server.Status != nova.StatusBuild {
+				break
+			}
+			// We don't want to flood the connection while polling the server waiting
+			// for it to start.
+			logger.Debugf("server has status %s, waiting 10 seconds before polling again...", server.Status)
+			time.Sleep(10 * time.Second)
```

Comment:
It's been taking between 15 to 21 seconds based on my sampling with a local openstack.  19 to 36 on a remote one.

The oracle provider waits for the instance to complete building as well.  

With more reading, I should add a timeout here, since an instance might get stuck in the build state.